tidy3d/components/simulation.py

```diff
@@ -4331,7 +4361,7 @@ def subsection(
             raise SetupError("Requested region does not intersect simulation domain")
 
         # restrict to the original simulation domain
-        new_bounds = Box.bounds_intersection(self.bounds, region.bounds)
+        new_bounds = Box.bounds_intersection(self.simulation_bounds, region.bounds)
```

Comment:
let's do:
```
if include_pml_cells:
    new_bounds = Box.bounds_intersection(self.simulation_bounds, region.bounds)
else:
    new_bounds = Box.bounds_intersection(self.bounds, region.bounds)
```
where `include_pml_cells: bool = False` is a new parameter. Then in mode solver we just call it with `include_pml_cells=True`

Reply:
What are the use cases that pml cells need to discarded? I think it is not necessary for mode solver?

Reply:
when I was initially designing this function the idea was that it should always return a subset of existing simulation. That is, in the limiting case of `new_sim = sim.subsection(region=td.Box(size=[inf, inf, inf]))` we get essentially the same simulation. A more practical example: when you have a large waveguide circuit and want to focus on one particular component, i.e., you can use something like `new_sim = sim.subsection(region=td.Box(size=(Lx, Ly, inf), center(xc, yc, 0))` without worrying that size in the z direction is accidentally increased